### ISSUE TITLE
Allow for non-existent event types in `Webhooks::Outgoing::Endpoint#event_types`

### ISSUE THREAD

**Issue by RichStone** [bullet-train-co/bullet_train-core#654]
It's possible that event types are created dynamically and deleted afterward. The below method raises an error in case the event type was deleted and not cleaned up on all the endpoints:

https://github.com/bullet-train-co/bullet_train-core/blob/5e6d042448abba11abe929c3c2ec11618a6301de/bullet_train-outgoing_webhooks/app/models/concerns/webhooks/outgoing/endpoint_support.rb#L36-L38

I'd propose to account for that case with an extra param like this:

```ruby
def event_types(ignore_deleted: false)
```

Because I don't think it can somehow be enforced that the deleting call cleans up related endpoints (I mean, it could, but I don't think we would want to go that route).

**Comment by jagthedrummer** [comment]
@RichStone I can see the potential for the error your described, but I'm curious how and why you would run into it. Like, can you describe a reproduction scenario where I could see it fail?

If we went with the `ignore_deleted` param that you mentioned, would we need to update other parts of the code to pass that param with different values at different times, or is this something you ran into with custom code?

Are there times when you would want it to raise an error if we encounter the scenario you describe? Or would it be reasonable to change that method so that it _never_ raises an error in that scenario? What if change the method to be this?

```ruby
def event_types 
  Webhooks::Outgoing::EventType.where(event_type_ids) 
end
```

**Comment by RichStone** [comment]
> @RichStone I can see the potential for the error your described, but I'm curious how and why you would run into it. Like, can you describe a reproduction scenario where I could see it fail?

Hey Jeremy, I think you should easily run into this if you:

1. Define an event type in event_types.yml.
2. Create a Webhooks::Outgoing::Endpoint with that type.
3. Remove that type.
4. Call Webhooks::Outgoing::Endpoint#event_types.

> If we went with the `ignore_deleted` param that you mentioned, would we need to update other parts of the code to pass that param with different values at different times, or is this something you ran into with custom code?
>
> Are there times when you would want it to raise an error if we encounter the scenario you describe? Or would it be reasonable to change that method so that it _never_ raises an error in that scenario? What if change the method to be this?
> 
> ```ruby
> def event_types 
>   Webhooks::Outgoing::EventType.where(event_type_ids) 
> end
> ```

I feel like that would be the correct way and elegant way to do this, but my old proposal had the `false` default to make it configurable and keep the old behavior for existing clients because I guessed that someone could rely on the raise to make "manual" clean up or something like that. But this might be far-fetched. In retrospect, I think in the code I encountered this, it would be the better solution, it doesn't seem like you'd want it to raise if you call it in one place and not raise in another, so no purpose to make it configurable.

Changing the data structure to an ActiveRecord::Relation shouldn't break things either.

**Comment by RichStone** [comment]
> Thanks for the details, @RichStone. Here's a PR to fix this issue: #983
> 
> Do you think there's anything else we should say in the description to help people understand the change?

The description looks great! Maybe it's also noteworthy that instead of an array this now returns an ActiveRecord::Relation?

**Comment by jagthedrummer** [comment]
Good idea to call that out, @RichStone. I updated the description on that PR. Thanks!
